### PR TITLE
Migrate to new `insert` and `search` endpoints for Embedding Indices

### PR DESCRIPTION
**Pull request by eob**
This PR:

* Converts all `index.insert` operations to ones that use the `Tag` object
* Converts all `index.search` operations to ones that return `Tag` objects
* removes testing for features that won't be supported in the new pluginInstance style interface
  - doing multiple searches at a time
  - listing the contents of an embedding index
  
Note:

* It looks like we'll need a very small change on the Engine to support this; the client & engine appear out of sync on data model with this -- DBSimdexRecord will need a tagId field, and the API endpoint will need to report it. I can make but wanted to flag for you @dkolas

### CODE REVIEW

File: src/steamship/data/block.py [hunk 1]
```diff
@@ -79,6 +79,11 @@ def query(
         )
         return res
 
+    def index(self, plugin_instance: Any = None):
+        """Index this block."""
+        tags = [Tag(text=self.text, file_id=self.id, block_id=self.id, kind="block")]
```
Comment (EniasCailliau) [inline src/steamship/data/block.py:84]: Flagging this to make sure this is intended: 

Do we want to assign the block_id to the file_id?

File: src/steamship/data/block.py [hunk 2]
```diff
@@ -79,6 +79,11 @@ def query(
         )
         return res
 
+    def index(self, plugin_instance: Any = None):
```
Comment (EniasCailliau) [inline src/steamship/data/block.py:82]:
```suggestion
    def index(self, embedding_plugin_instance: EmbeddingIndex = None):
```

This only works for embedding_plugin_instances right?

**Review by EniasCailliau — APPROVED** [review]
LGTM

File: src/steamship/data/block.py (hunk 1 above)
Comment (eob) [reply src/steamship/data/block.py:84]: Yikes nice catch @EniasCailliau !

File: src/steamship/data/block.py (hunk 2 above)
Comment (eob) [reply src/steamship/data/block.py:82]: Yes -- will update w/ this suggestion. Thanks!

**Comment by eob** [comment]
Tests green against staging so merging.

**Pull request merged**